Initial implementation of gRPC reflection (#1)
shaneutt/rust-grpc-demo: README.md

[protoc]:https://wikipedia.org/wiki/Protocol_Buffers

> ## Step 3: Compiling Protobuf

Reviewer: Up in the pre-requisites section, should we add a 

```
> **Warning**
```

For Mac users to let them know they'll need to use a nightly build of rust due to differences with dependencies in Linux?

e.g. the `rustix` crate had some sort of unstable feature usage on your Mac that it didn't on my Linux system:

```
> **Warning**: If you're using a Mac to run this demo, some of the dependencies here require unstable features
> on Mac that aren't required on Linux. You'll want to ensure you have a nightly build of Rust installed with
> `rustup toolchain add nightly` and anywhere you see `cargo build` you'll need to run `cargo +nightly build`
> instead.
```

Maybe that?

Author: Actuality, with provided commands (release compilation) I couldn't reproduce the issue.

Reviewer: Ah good!

$ cargo run --release --bin server
```

At first, we are going to test our server using [Insomnia][insomnia].

> Insomnia is a free API testing tool, and it supports testing gRPC services.

Reviewer: For the crowd interested in this blog post, I would suggest we make sure to refer to it as OSS:

```suggestion
Insomnia is an open source API testing tool, and it supports testing gRPC services.
```

> After you downloaded and installed Insomnia, open it and click ⨁ icon and select «gRPC Request».

Reviewer: Perhaps we should add this to the pre-requisites?

![grpc_request.png](images%2Fgrpc_request.png)

Enter `localhost:9001` in gRPC server bar. 
In gRPC method dropdown, select _Click to use server reflection_.

![server_reflection.png](images%2Fserver_reflection.png)

Select Unary method `/Inventory/Add` and add body of json request. 

```json
{
   "identifier": {
      "sku": "TESTSKU"
   },
   "stock": {
      "price": 1.9900000095367432,
      "quantity": 20
   },
   "information": {
      "name": "bananas",
      "description": "yellow fruit"
   }
}
```

![inso.gif](images%2Finso.gif)

> **Note**: Feel free to experiment with other methods of Inventory service on your own.

> Now let's test gRPC client using cli app we developed earlier.

Reviewer: It seems like this whole section should be _before_ we start building the client, after we build the server no?

like a step specifically just to make sure the server is working, then we can go on and add the CLI?

Because at this point, we already have a tool that can do everything we're demonstrating here so it almost feels a bit redundant, but as a tool to initially test that our service is building and running correctly it makes sense, wdyt?
